### DISCUSSION TITLE
paginate->withQueryString()

### DISCUSSION THREAD

**Discussion by kevnk** [statamic-rad-pack/runway#348] [Help]
I'm running the paginate with additional query params like `?category=family&sort=new_releases` and going to the next page replaces them with `?page=2` and the user loses their filtering. This can be easily solved with [Laravel's method `withQueryString`](https://laravel.com/docs/10.x/pagination#appending-query-string-values). I'm not sure you want to do this by default or if you want to add another param to the RunwayTag.

I can't think of any CONs to having it in there by default, but maybe there's a slight performance hit? 

I'm happy to send a pull request. 

Thanks!

**Comment by duncanmcclean** [comment]
Closing in favour of the issue (this sounds like a bug, not a feature): https://github.com/duncanmcclean/runway/issues/349.
